include/mbedtls/config.h

```diff
@@ -3688,6 +3688,4 @@
 #include MBEDTLS_USER_CONFIG_FILE
 #endif
 
-#include "mbedtls/check_config.h"
```

Comment:
[There's one place where `check_config.h` _modifies_ the configuration](https://github.com/ARMmbed/mbedtls/blob/mbedtls-2.22.0/include/mbedtls/check_config.h#L47).

```
#if defined(_WIN32)
#if !defined(MBEDTLS_PLATFORM_C)
#error "MBEDTLS_PLATFORM_C is required on Windows"
#endif

/* Fix the config here. Not convenient to put an #ifdef _WIN32 in config.h as
 * it would confuse config.py. */
#if !defined(MBEDTLS_PLATFORM_SNPRINTF_ALT) && \
    !defined(MBEDTLS_PLATFORM_SNPRINTF_MACRO)
#define MBEDTLS_PLATFORM_SNPRINTF_ALT
#endif

#if !defined(MBEDTLS_PLATFORM_VSNPRINTF_ALT) && \
    !defined(MBEDTLS_PLATFORM_VSNPRINTF_MACRO)
#define MBEDTLS_PLATFORM_VSNPRINTF_ALT
#endif
#endif /* _WIN32 */
```

If `check_config.h` is no longer included from `config.h`, and thus only included from library code and not from application code, this won't do. Application code and library code must have the same configuration.

Reply:
The only code in the library that is functionally affected by `MBEDTLS_PLATFORM_SNPRINTF_ALT` or `MBEDTLS_PLATFORM_VSNPRINTF_ALT` is in `platform.h` and `platform.c`. Application code that is affected by these macros can be reasonably expected to include `mbedtls/platform.h`. Therefore I think this hack can be adapted to fit in `platform.h` instead.

Feature detection (`version_features.c`, `query_config`) would no longer report these alt symbols correctly. I think that an acceptable change would be to not set the symbols, but cause their effect in `platform.h` regardless.

Reply:
Fortunately, it turns out that changing the configuration is not necessary: `platform.h` will do its thing (activate an snprintf wrapper) regardless. https://github.com/ARMmbed/mbedtls/pull/3453